### PR TITLE
PROGRAMMES-5802 aggregated_galleries_count

### PR DESCRIPTION
**Pull request by Tasiobg**
https://jira.dev.bbc.co.uk/browse/PROGRAMMES-5802
Depends on https://github.com/bbc/programmes-faucet/pull/29

- update and rename variables and functions for "available galleries count" to "aggregated galleries count"
- Add aggregatedGalleriesCount to Clips
- unitest

### CODE REVIEW

File: src/Data/ProgrammesDb/Entity/Clip.php [hunk 1]
```diff
@@ -2,11 +2,13 @@
...
+use BBC\ProgrammesPagesService\Data\ProgrammesDb\Entity\EntityInterfaces\GalleriesCountableInterface;
 use Doctrine\ORM\Mapping as ORM;
 
 /**
  * @ORM\Entity()
  */
-class Clip extends ProgrammeItem
+class Clip extends ProgrammeItem implements GalleriesCountableInterface
 {
+    use Traits\AggregatedGalleriesCountMethodsTrait;
```
Comment (votemike) [inline src/Data/ProgrammesDb/Entity/Clip.php]: Can this be imported?

File: src/Domain/Entity/Episode.php [hunk 2]
```diff
@@ -18,7 +18,7 @@ class Episode extends ProgrammeItem
     private $availableClipsCount;
 
     /** @var int */
-    private $availableGalleriesCount;
+    private $aggregatedGalleriesCount;
```
Comment (votemike) [inline src/Domain/Entity/Episode.php]: Would it not be better to put all of these changes in `ProgrammeItem`?

File: src/Domain/Entity/ProgrammeContainer.php [hunk 3]
```diff
@@ -21,7 +21,7 @@
     private $availableEpisodesCount;
 
     /** @var int */
-    private $availableGalleriesCount;
+    private $aggregatedGalleriesCount;
```
Comment (votemike) [inline src/Domain/Entity/ProgrammeContainer.php]: Would it not be better to put all of these changes in `Programme`?

File: src/Data/ProgrammesDb/Entity/Clip.php (hunk 1 above)
Comment (Tasiobg) [reply src/Data/ProgrammesDb/Entity/Clip.php]: We are using traits like that in few places, I think importing them is a more elegant, thanks.

File: src/Domain/Entity/ProgrammeContainer.php (hunk 3 above)
Comment (Tasiobg) [reply src/Domain/Entity/ProgrammeContainer.php]: Looks like the aggregatedGalleriesCount code can be moved inside Programme, thanks.

**Pull request merged**